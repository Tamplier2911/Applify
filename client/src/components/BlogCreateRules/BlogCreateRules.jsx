// import "./BlogCreateRules.scss";
import React from "react";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectCurrentLanguage } from "../../redux/lang/lang.selectors";

// JS Rendering CSS
import {
  BlogCreateRulesContainer,
  BlogCreateRulesTitle,
  BlogCreateRulesUl,
  BlogCreateRulesLi,
  BlogCreateRulesLiSelector,
  BlogCreateRulesLiDescriptor,
  BlogCreateRulesCode,
  BlogCreateRulesSection,
  BlogCreateRulesCodeSelector,
  BlogCreateRulesCodeSample,
} from "./BlogCreateRulesStyles";

// component constants
import blogCreateRulesData from "./BlogCreateRulesConstants";

const BlogCreateRules = ({ lang }) => {
  const {
    blogRulesTitles,
    blogRulesSelectors,
    blogRulesDescriptors,
    blogRulesCode,
  } = blogCreateRulesData[lang];
  return (
    <BlogCreateRulesContainer data-test="blog-create-rules">
      <BlogCreateRulesTitle>{blogRulesTitles[0]}</BlogCreateRulesTitle>
      <BlogCreateRulesUl>
        <BlogCreateRulesLi>
          <BlogCreateRulesLiSelector>
            {blogRulesSelectors[0]}
          </BlogCreateRulesLiSelector>
          <BlogCreateRulesLiDescriptor>
            {blogRulesDescriptors[0]}
          </BlogCreateRulesLiDescriptor>
        </BlogCreateRulesLi>
        <BlogCreateRulesLi>
          <BlogCreateRulesLiSelector>
            {blogRulesSelectors[1]}
          </BlogCreateRulesLiSelector>
          <BlogCreateRulesLiDescriptor>
            {blogRulesDescriptors[1]}
          </BlogCreateRulesLiDescriptor>
        </BlogCreateRulesLi>
        <BlogCreateRulesLi>
          <BlogCreateRulesLiSelector>
            {blogRulesSelectors[2]}
          </BlogCreateRulesLiSelector>
          <BlogCreateRulesLiDescriptor>
            {blogRulesDescriptors[2]}
          </BlogCreateRulesLiDescriptor>
        </BlogCreateRulesLi>
        <BlogCreateRulesLi>
          <BlogCreateRulesLiSelector>
            {blogRulesSelectors[3]}
          </BlogCreateRulesLiSelector>
          <BlogCreateRulesLiDescriptor>
            {blogRulesDescriptors[3]}
          </BlogCreateRulesLiDescriptor>
        </BlogCreateRulesLi>
        <BlogCreateRulesLi>
          <BlogCreateRulesLiSelector>
            {blogRulesSelectors[4]}
          </BlogCreateRulesLiSelector>
          <BlogCreateRulesLiDescriptor>
            {blogRulesDescriptors[4]}
          </BlogCreateRulesLiDescriptor>
        </BlogCreateRulesLi>
        <BlogCreateRulesLi>
          <BlogCreateRulesLiSelector>
            {blogRulesSelectors[5]}
          </BlogCreateRulesLiSelector>
          <BlogCreateRulesLiDescriptor>
            {blogRulesDescriptors[5]}
          </BlogCreateRulesLiDescriptor>
        </BlogCreateRulesLi>
      </BlogCreateRulesUl>
      <BlogCreateRulesTitle>{blogRulesTitles[1]}</BlogCreateRulesTitle>
      <BlogCreateRulesCode>
        <BlogCreateRulesSection>
          <BlogCreateRulesCodeSelector>
            {blogRulesSelectors[0]}
          </BlogCreateRulesCodeSelector>
        </BlogCreateRulesSection>
        <BlogCreateRulesSection>
          <BlogCreateRulesCodeSelector>
            {blogRulesSelectors[1]}
          </BlogCreateRulesCodeSelector>
          <BlogCreateRulesCodeSample>
            {blogRulesCode[0]}
          </BlogCreateRulesCodeSample>
          <BlogCreateRulesCodeSelector>
            {blogRulesSelectors[1]}
          </BlogCreateRulesCodeSelector>
        </BlogCreateRulesSection>
        <BlogCreateRulesSection>
          <BlogCreateRulesCodeSelector>
            {blogRulesSelectors[2]}
          </BlogCreateRulesCodeSelector>
          <BlogCreateRulesCodeSample>
            {blogRulesCode[1]}
          </BlogCreateRulesCodeSample>
          <BlogCreateRulesCodeSelector>
            {blogRulesSelectors[2]}
          </BlogCreateRulesCodeSelector>
        </BlogCreateRulesSection>
        <BlogCreateRulesSection>
          <BlogCreateRulesCodeSelector>
            {blogRulesSelectors[3]}
          </BlogCreateRulesCodeSelector>
          <BlogCreateRulesCodeSample>
            {blogRulesCode[2]}
          </BlogCreateRulesCodeSample>
          <BlogCreateRulesCodeSelector>
            {blogRulesSelectors[3]}
          </BlogCreateRulesCodeSelector>
        </BlogCreateRulesSection>
        <BlogCreateRulesSection>
          <BlogCreateRulesCodeSelector>
            {blogRulesSelectors[4]}
          </BlogCreateRulesCodeSelector>
          <BlogCreateRulesCodeSample>
            {blogRulesCode[3]}
          </BlogCreateRulesCodeSample>
          <BlogCreateRulesCodeSelector>
            {blogRulesSelectors[4]}
          </BlogCreateRulesCodeSelector>
        </BlogCreateRulesSection>
        <BlogCreateRulesSection>
          <BlogCreateRulesCodeSelector>
            {blogRulesSelectors[4]}
          </BlogCreateRulesCodeSelector>
          <BlogCreateRulesCodeSample>
            {blogRulesCode[4]}
          </BlogCreateRulesCodeSample>
          <BlogCreateRulesCodeSelector>
            {blogRulesSelectors[4]}
          </BlogCreateRulesCodeSelector>
        </BlogCreateRulesSection>
        <BlogCreateRulesSection>
          <BlogCreateRulesCodeSelector>
            {blogRulesSelectors[5]}
          </BlogCreateRulesCodeSelector>
          <BlogCreateRulesCodeSample>
            {blogRulesCode[5]}
          </BlogCreateRulesCodeSample>
          <BlogCreateRulesCodeSelector>
            {blogRulesSelectors[5]}
          </BlogCreateRulesCodeSelector>
        </BlogCreateRulesSection>
        <BlogCreateRulesCodeSelector>
          {blogRulesSelectors[0]}
        </BlogCreateRulesCodeSelector>
      </BlogCreateRulesCode>
    </BlogCreateRulesContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  lang: selectCurrentLanguage,
});

export default connect(mapStateToProps)(BlogCreateRules);
