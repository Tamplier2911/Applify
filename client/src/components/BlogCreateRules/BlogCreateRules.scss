/*

.blogcreate-rules {
  display: grid;
  grid-row-gap: 2rem;
  font-family: var(--font-title);
  padding: 1rem;
  margin-bottom: 4rem;

  background-color: var(--cl-contacts-bg);
  box-shadow: 0rem 0.1rem 0.6rem var(--cl-square-shadow);

  &__title {
    font-size: 2.2rem;
    text-align: center;
  }

  &__ul {
    list-style: none;
  }

  &__li {
    display: grid;
    grid-template-columns: 1fr 9fr;

    @media only screen and (max-width: 872px) {
      grid-template-columns: 1fr 8fr;
    }

    @media only screen and (max-width: 720px) {
      grid-template-columns: 1fr 5fr;
    }

    @media only screen and (max-width: 500px) {
      grid-template-columns: 1fr 3fr;
    }

    @media only screen and (max-width: 310px) {
      grid-template-columns: 1fr;
    }

    &--selector {
      color: var(--cl-github);
    }
    &--desc {
    }
  }

  &__code {
    display: grid;
    &--section {
      display: grid;
      grid-template-columns: repeat(3, min-content);

      @media only screen and (max-width: 425px) {
        grid-template-columns: 1fr;
      }
    }
    &--selector {
      color: var(--cl-github);
    }
    &--sample {
      white-space: nowrap;

      @media only screen and (max-width: 425px) {
        white-space: unset;
      }

      @media only screen and (max-width: 245px) {
        overflow: hidden;
        text-overflow: ellipsis;
      }
    }
  }
}

*/
