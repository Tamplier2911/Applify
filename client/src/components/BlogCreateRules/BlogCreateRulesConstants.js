const blogCreateRulesData = {
  eng: {
    blogRulesTitles: [
      "Simple rules on how to create a blog, with data to be parsed in right way:",
      "Example:"
    ],
    blogRulesSelectors: ["%BLOCK%", "%H%", "%P%", "%A%", "%L%", "%CODE%"],
    blogRulesDescriptors: [
      "- wrap around every block of content.",
      "- wrap around anything meant to be a header.",
      "- wrap around any paragraph.",
      "- wrap around any anchor tag.",
      "- wrap around any list item.",
      "- wrap around any code snippet."
    ],
    blogRulesCode: [
      "This is a header!",
      "This is some paragraph that mean to be rendered.",
      "https://www.google.com:Google.com",
      "List item - 1",
      "List item - 2",
      "console.log('Hello, world!')"
    ]
  },
  rus: {
    blogRulesTitles: [
      "Простые правила - как создать блог с данными, для правильного анализа:",
      "Образец:"
    ],
    blogRulesSelectors: ["%BLOCK%", "%H%", "%P%", "%A%", "%L%", "%CODE%"],
    blogRulesDescriptors: [
      "- обернуть вокруг каждого блока контента.",
      "- обернуть все, что должно быть заголовком.",
      "- обернуть любой абзац.",
      "- обернуть вокруг тега привязки.",
      "- обернуть вокруг любого элемента списка.",
      "- обернуть любой фрагмент кода."
    ],
    blogRulesCode: [
      "Это заголовок!",
      "Это какой-то параграф.",
      "https://www.google.com:Google.com",
      "Элемент списка - 1",
      "Элемент списка - 2",
      "console.log('Hello, world!')"
    ]
  },
  ukr: {
    blogRulesTitles: [
      "Прості правила - як створити блог з даними, для правильного аналізу:",
      "Зразок:"
    ],
    blogRulesSelectors: ["%BLOCK%", "%H%", "%P%", "%A%", "%L%", "%CODE%"],
    blogRulesDescriptors: [
      "- обернути навколо кожного блоку контенту.",
      "- обернути все, що повинно бути заголовком.",
      "- обернути абзац.",
      "- обернути навколо тега прив'язки..",
      "- обернути навколо будь-якого елементу списку.",
      "- обернути будь-який фрагмент коду."
    ],
    blogRulesCode: [
      "Це заголовок!",
      "Це якийсь параграф.",
      "https://www.google.com:Google.com",
      "Елементу списку - 1",
      "Елементу списку - 2",
      "console.log('Hello, world!')"
    ]
  }
};

export default blogCreateRulesData;
