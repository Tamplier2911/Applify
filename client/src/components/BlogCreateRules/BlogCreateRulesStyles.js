import styled from "styled-components";

export const BlogCreateRulesContainer = styled.div`
  display: grid;
  grid-row-gap: 2rem;
  font-family: var(--font-title);
  padding: 1rem;
  margin-bottom: 4rem;

  background-color: var(--cl-contacts-bg);
  box-shadow: 0rem 0.1rem 0.6rem var(--cl-square-shadow);
`;

export const BlogCreateRulesTitle = styled.h3`
  font-size: 2.2rem;
  text-align: center;
  font-weight: 400;
`;

export const BlogCreateRulesUl = styled.ul`
  list-style: none;
`;

export const BlogCreateRulesLi = styled.li`
  display: grid;
  grid-template-columns: 1fr 9fr;

  @media only screen and (max-width: 872px) {
    grid-template-columns: 1fr 8fr;
  }

  @media only screen and (max-width: 720px) {
    grid-template-columns: 1fr 5fr;
  }

  @media only screen and (max-width: 500px) {
    grid-template-columns: 1fr 3fr;
  }

  @media only screen and (max-width: 310px) {
    grid-template-columns: 1fr;
  }
`;

export const BlogCreateRulesLiSelector = styled.span`
  color: var(--cl-github);
`;

export const BlogCreateRulesLiDescriptor = styled.span``;

export const BlogCreateRulesCode = styled.div`
  display: grid;
`;

export const BlogCreateRulesSection = styled.div`
  display: grid;
  grid-template-columns: repeat(3, min-content);

  @media only screen and (max-width: 425px) {
    grid-template-columns: 1fr;
  }
`;

export const BlogCreateRulesCodeSelector = styled.div`
  color: var(--cl-github);
`;

export const BlogCreateRulesCodeSample = styled.div`
  white-space: nowrap;

  @media only screen and (max-width: 425px) {
    white-space: unset;
  }

  @media only screen and (max-width: 245px) {
    overflow: hidden;
    text-overflow: ellipsis;
  }
`;
