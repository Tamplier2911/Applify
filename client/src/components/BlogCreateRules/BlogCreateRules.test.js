import React from "react";
import toJson from "enzyme-to-json";

// component
import BlogCreateRules from "./BlogCreateRules";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<BlogCreateRules />", () => {
  const initialStore = {
    lang: { languagePanelHidden: true, currentLanguage: "eng" },
  };

  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<BlogCreateRules store={store} />);
  });

  it("renders without error", () => {
    const blogCreateRules = findByTestAttr(wrapper, "blog-create-rules");
    expect(blogCreateRules.length).toBe(1);
  });

  it("matches snapshot", () => {
    const blogCreateRules = findByTestAttr(wrapper, "blog-create-rules");
    expect(toJson(blogCreateRules)).toMatchSnapshot();
  });
});
