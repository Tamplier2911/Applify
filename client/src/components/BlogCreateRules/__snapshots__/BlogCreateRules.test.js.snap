// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogCreateRules /> matches snapshot 1`] = `ShallowWrapper {}`;
